src/CoreWCF.ConfigurationManager/src/CoreWCF/Configuration/ConfigurationChannelBindingUtility.cs

```diff
@@ -0,0 +1,260 @@
+            else
+            {
+                ServiceNameCollection merged = primaryList.Merge(subset);
+
+                //The merge routine only adds an entry if it is unique.
+                result = (merged.Count == primaryList.Count);
+            }
+
+            return result;
+        }
+
+        public static void Dispose(ref ChannelBinding channelBinding)
+        {
+            // Explicitly cast to IDisposable to avoid the SecurityException.
+            IDisposable disposable = (IDisposable)channelBinding;
+            channelBinding = null;
+            if (disposable != null)
+            {
+                disposable.Dispose();
+            }
+        }
+
+        private class DuplicatedChannelBinding : ChannelBinding
+        {
+            [SecurityCritical]
+            int _size;
+
+            DuplicatedChannelBinding()
+            {
+
+            }
+
+            public override int Size
+            {
+                [SecuritySafeCritical]
+                get { return this._size; }
+            }
+
+            
+            [SecuritySafeCritical]
+            internal static ChannelBinding CreateCopy(ChannelBinding source)
+            {
+                Fx.Assert(source != null, "source ChannelBinding should have been checked for null previously");
+
+                if (source.IsInvalid || source.IsClosed)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ObjectDisposedException(source.GetType().FullName));
+                }
+
+                if (source.Size <= 0)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException("source.Size", source.Size,
+                        SR.Format(SR.ValueMustBePositive)));
+                }
+
+                //Instantiate the SafeHandle before trying to allocate the native memory
+                DuplicatedChannelBinding duplicate = new DuplicatedChannelBinding();
+
+                //allocate the native memory and make a deep copy of the original.
+                duplicate.Initialize(source);
+
+                return duplicate;
+            }
+
+            [SecurityCritical]
+            private unsafe void Initialize(ChannelBinding source)
+            {
+                //allocates the memory pointed to by this.handle
+                //and sets this.size after allocation succeeds.
+                AllocateMemory(source.Size);
+
+                byte* sourceBuffer = (byte*)source.DangerousGetHandle().ToPointer();
+                byte* destinationBuffer = (byte*)this.handle.ToPointer();
+
+                for (int i = 0; i < source.Size; i++)
+                {
+                    destinationBuffer[i] = sourceBuffer[i];
+                }
+
+                this._size = source.Size;
+            }
+
+            [ReliabilityContract(Consistency.WillNotCorruptState, Cer.Success)]
+            private void AllocateMemory(int bytesToAllocate)
```

Comment:
The constrained execution feature (CER) isn't available on core. Remove this attribute. Also remove call to `RuntimeHelpers.PrepareConstrainedRegions();` and remove the try/finally block from around the SetHandle call as it's also there as part of CER. Also remove the comment as it's not applicable any more as we don't have unloadable app domains.